data-processing/pulsar-flink/2.7.0/pulsar-flink.md

```diff
@@ -200,6 +200,49 @@ In some scenarios, users need messages to be strictly guaranteed message order t
 
 Pulsar Flink connector supports this feature as well. This feature can be enabled by configuring the `enable-key-hash-range=true` parameter. When enabled, the range of Key Hash processed by each consumer is divided based on the parallelism of the task.
 
+## Fault tolerance
+
+Pulsar Flink connector 2.7.0 provides different semantics for source and sink.
+
+### Source
+
+For Pulsar source, Pulsar Flink connector 2.7.0 provides `exactly-once` semantic.
+
+### Sink
+
+Pulsar Flink connector 2.4.12 only supports `at-least-once` semantic for sink. Based on transactions supported in Pulsar 2.7.0 and the Flink [`TwoPhaseCommitSinkFunction` API](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/streaming/api/functions/sink/TwoPhaseCommitSinkFunction.html), Pulsar Flink connector 2.7.0 supports both `exactly-once` and `at-least-once` semantics for sink. For more information, see [here](https://flink.apache.org/2021/01/07/pulsar-flink-connector-270.html).
+
+Before setting `exactly_once` semantic for a sink, you need to make the following configuration changes.
+
+1. In Pulsar, transaction related functions are **disabled by default**. In this case, you need to set `transactionCoordinatorEnabled = true` in the configuration file (`conf/standalone.conf` or `conf/broker.conf`) .
```

Comment:
I think we can use an unordered list instead of an ordered list. It seems these two steps have no required sequence.

Reply:
Users need to perform step 1 and then go with step 2.